content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -1057,17 +1087,42 @@ extra:
   - profile
 ```
 
+{{< note >}}
+Prior to Kubernetes 1.11.3 (and 1.10.7, 1.9.11), the `extra` key could only contain characters that
+were [legal in HTTP header labels](https://tools.ietf.org/html/rfc7230#section-3.2.6).
+{{< /note >}}
+
+#### Client certificate {#reverse-proxy-client-certificate}
+
 In order to prevent header spoofing, the authenticating proxy is required to present a valid client
 certificate to the API server for validation against the specified CA before the request headers are
-checked. WARNING: do **not** reuse a CA that is used in a different context unless you understand
+checked.
+
+<!-- deliberately repeating the earlier warning -->
+Do **not** reuse a CA that is used in a different context unless you understand
 the risks and the mechanisms to protect the CA's usage.
 
-* `--requestheader-client-ca-file` Required. PEM-encoded certificate bundle. A valid client certificate
-  must be presented and validated against the certificate authorities in the specified file before the
-  request headers are checked for user names.
-* `--requestheader-allowed-names` Optional. List of Common Name values (CNs). If set, a valid client
-  certificate with a CN in the specified list must be presented before the request headers are checked
-  for user names. If empty, any CN is allowed.
+### Static token file integration {#static-token-file}
+
+The API server reads static bearer tokens from a file when given the `--token-auth-file=<SOMEFILE>`
+option on the command line.
+In Kubernetes {{< skew currentVersion >}}, tokens last indefinitely, and the token list cannot be
+changed without restarting the API server.
+
+The token file is a CSV file with a minimum of 3 columns: token, user name, user uid,
+followed by a comma-separated list of optional group names.
+
+
+{{< note >}}
+If you have more than one group, the column must be double quoted e.g.
+
+```conf
+token,user,uid,"group1,group2,group3"
+```
+{{< /note >}}
+
+The Kubernetes project recommends using [webhook token authenticator](#webhook-token-authentication)
+instead of this mechanism.
```

Comment:
rather than a blanket redirect to token webhook, I think it would be better to describe what this is or is not good for

* It is good for tokens that are permanent or long-lived and do not need to be revoked or rotated
* It is good for credentials used by agents local to a single kube-apiserver, especially ones that only need read access for monitoring purposes, or during cluster startup, when a self-hosted webhook token authenticator would not yet be functional